python/physicsobjects/DiLepton.py

```diff
@@ -137,21 +135,35 @@ def cosTransversePointingAngleBS(self):
-                          '\t deltaEta %.5f' %self.deta(),
-                          '\t vertex x=%.2f y=%.2f z=%.2f' %(self.vtx().x(), self.vtx().y(), self.vtx().z()),
-                          '\t vertex chi2 %.5f' %self.chi2(),
-                          '\t 2d displacement from BS %.3f with L/sigma %.3f' %(self.disp2DFromBS(), self.disp2DFromBSSignificance()),
-                          '\t 2d displacement from PV %.3f with L/sigma %.3f' %(self.disp2DFromPV(), self.disp2DFromPVSignificance()),
-                          '\t 3d displacement from PV %.3f with L/sigma %.3f' %(self.disp3DFromPV(), self.disp3DFromPVSignificance()),
-                          '\t pointing angle cosine %.7f' %self.cosTransversePointingAngleBS(),
-                          '',
-                        ])                          
\ No newline at end of file
+        try:
+            return '\n'.join(['',
+                              self.lep1().__str__(),
+                              self.lep2().__str__(),
+                              '\t pt       %.2f' %self.pt(),
+                              '\t eta      %.2f' %self.eta(),
+                              '\t phi      %.2f' %self.phi(),
+                              '\t mass     %.3f' %self.mass(),
+                              '\t deltaR   %.5f' %self.dr(),
+                              '\t deltaPhi %.5f' %self.dphi(),
+                              '\t deltaEta %.5f' %self.deta(),
+                              '\t vertex x=%.2f y=%.2f z=%.2f' %(self.vtx().x(), self.vtx().y(), self.vtx().z()),
+                              '\t vertex chi2 %.5f' %self.chi2(),
+                              '\t 2d displacement from BS %.3f with L/sigma %.3f' %(self.disp2DFromBS(), self.disp2DFromBSSignificance()),
+                              '\t 2d displacement from PV %.3f with L/sigma %.3f' %(self.disp2DFromPV(), self.disp2DFromPVSignificance()),
+                              '\t 3d displacement from PV %.3f with L/sigma %.3f' %(self.disp3DFromPV(), self.disp3DFromPVSignificance()),
+                              '\t pointing angle cosine %.7f' %self.cosTransversePointingAngleBS(),
+                              '',
+                            ])                          
+        except:
+            return '\n'.join(['',
+                              self.lep1().__str__(),
+                              self.lep2().__str__(),
+                              '\t pt       %.2f' %self.pt(),
+                              '\t eta      %.2f' %self.eta(),
+                              '\t phi      %.2f' %self.phi(),
+                              '\t mass     %.3f' %self.mass(),
+                              '\t deltaR   %.5f' %self.dr(),
+                              '\t deltaPhi %.5f' %self.dphi(),
+                              '\t deltaEta %.5f' %self.deta(),
+                              '',
+                            ])                          
```

Comment:
why `try / except`?

Reply:
this is also an uncleaned and finished construction site. thanks for pointing it out. I removed the try/except